code/datums/components/mood.dm

```diff
@@ -199,6 +199,12 @@
 		if(prob(0.05))
 			add_event(null, "jolly", /datum/mood_event/jolly)
 			clear_event(null, "depression")
+	if(owner.has_trait(TRAIT_WEALTHY))
+		if(prob(0.02))
+			var/mob/living/carbon/human/H = quirk_holder
```

Comment:
Only humans can get this quirk?

Reply:
huh?

Reply:
Isnt living/carbon/human to check for humans? Shouldn't it be only /living?

Reply:
Any living mob can have a mood component according to the checks above.

Same is true with quirks since the quirk_owner is typecasted as mob/living. Cringe.

Reply:
In short tho, it's wrong to assume the owner of the component (dunno why quirk_holder is here) is human.

Reply:
Oh, ok, Sorry. I dont use many datums

Reply:
you were right in asking.